repose-aggregator/components/filters/uri-stripper-filter/src/test/scala/org/openrepose/filters/uristripper/UriStripperFilterRequestTest.scala

```diff
@@ -0,0 +1,1028 @@
+
+import scala.xml.XML
+
+@RunWith(classOf[JUnitRunner])
+class UriStripperFilterRequestTest extends FunSpec with BeforeAndAfterEach with Matchers with MockitoSugar {
+
+  import UriStripperFilterRequestTest._
+
+  var filter: UriStripperFilter = _
+  var request: MockHttpServletRequest = _
+  var response: MockHttpServletResponse = _
+  var filterChain: FilterChain = _
+
+  override def beforeEach() = {
+    request = new MockHttpServletRequest
+    response = new MockHttpServletResponse
+    filterChain = mock[FilterChain]
+    filter = new UriStripperFilter(null)
+  }
+
+  describe("request links handling") {
+    describe("json") {
+      it("should not alter the body if the uri does not match the configured regex") {
+        val config =
+          """<?xml version="1.0" encoding="UTF-8"?>
+            |<uri-stripper xmlns="http://docs.openrepose.org/repose/uri-stripper/v1.0" rewrite-location="false" token-index="1">
+            |    <link-resource uri-path-regex="/v1/[^/]+/bar">
+            |        <request>
+            |            <json>$.link</json>
+            |        </request>
+            |    </link-resource>
+            |</uri-stripper>
+          """.stripMargin
+
+        val requestBody =
+          """{
+            |  "link": "http://example.com/v1/12345/bar"
+            |}
+          """.stripMargin
+
+        filter.configurationUpdated(Marshaller.uriStripperConfigFromString(config))
+        request.setRequestURI("/v1/12345/foo")
+        request.setContent(requestBody.getBytes(CHARSET_UTF8))
+        request.setContentType(MimeType.APPLICATION_JSON.toString)
+
+        filter.doFilter(request, response, filterChain)
+
+        getPostFilterRequestBody shouldEqual requestBody
+      }
+
+      it("should not alter the body if the method does not match the configured method") {
+        val config =
+          """<?xml version="1.0" encoding="UTF-8"?>
+            |<uri-stripper xmlns="http://docs.openrepose.org/repose/uri-stripper/v1.0" rewrite-location="false" token-index="1">
+            |    <link-resource uri-path-regex=".*" http-methods="POST">
+            |        <request>
+            |            <json>$.link</json>
+            |        </request>
+            |    </link-resource>
+            |</uri-stripper>
+          """.stripMargin
+
+        val requestBody =
+          """{
+            |  "link": "http://example.com/v1/12345/foo"
+            |}
+          """.stripMargin
+
+        filter.configurationUpdated(Marshaller.uriStripperConfigFromString(config))
+        request.setMethod("GET")
+        request.setRequestURI("/v1/12345/foo")
+        request.setContent(requestBody.getBytes(CHARSET_UTF8))
+        request.setContentType(MimeType.APPLICATION_JSON.toString)
+
+        filter.doFilter(request, response, filterChain)
+
+        getPostFilterRequestBody shouldEqual requestBody
+      }
+
+      it("should not alter the body if the method does not match the configured resource type") {
```

Comment:
Maybe a name that better draws attention to the fact that the config is no longer on request, it took me a minute to figure out why this wouldn't fail.